src/NuGet.Configuration/PackageSource/PackageSourceProvider.cs

```diff
@@ -84,7 +83,7 @@ private IEnumerable<PackageSource> LoadConfigurationDefaultSources(IEnumerable<P
         {
 #if !DNXCORE50
             // Global default NuGet source doesn't make sense on Mono
-            if (_isMono.Value)
+            if (RuntimeEnvironmentHelper.IsMono)
```

Comment:
Not sure if there is a really good way to do it, but I assume that if you are on *nix you can rely on the pathseparator being forward slash. Alternatively the answer from http://stackoverflow.com/questions/430256/how-do-i-determine-whether-the-filesystem-is-case-sensitive-in-net is pretty heavy handed but can be done. Note that it is not accurate because it depends on the folder you are in.